src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -201,7 +202,8 @@ public TopicPartitionWriter(
         config.getString(StorageSinkConnectorConfig.SCHEMA_COMPATIBILITY_CONFIG));
 
     String logsDir = config.getLogsDirFromTopic(tp.topic());
-    wal = storage.wal(logsDir, tp);
+    boolean disableWal = config.getBoolean(HdfsSinkConnectorConfig.DISABLE_WAL_CONFIG);
+    wal = disableWal ? new NoopWAL() : storage.wal(logsDir, tp);
```

Comment:
Could we have used the memoryWAL?

Reply:
It looks like it might have worked - because `getLatestOffset()` returns null. But I think this might be clearer that there are no side affects and things are really disabled.